### PR TITLE
[PIL-1363-nil-return] Fix nil return response

### PR DESCRIPTION
**Pull request by ridow-hmrc**
Fix nilreturn response, which should be the same as liability return

### CODE REVIEW

File: app/uk/gov/hmrc/pillar2stubs/controllers/UKTRSubmitController.scala [hunk 1]
```diff
@@ -45,41 +46,51 @@ class UKTRSubmitController @Inject() (
...
-                          "processingDate" -> ZonedDateTime.now(ZoneId.of("UTC")),
-                          "code"           -> "001",
-                          "text"           -> "Invalid date range: accountingPeriodTo must be after accountingPeriodFrom"
+            if (!submission.accountingPeriodValid) {
+              logger.error("Invalid date range: accountingPeriodTo is before accountingPeriodFrom")
+              Future.successful(
+                UnprocessableEntity(
+                  Json.obj(
+                    "errors" -> Json.obj(
+                      "processingDate" -> ZonedDateTime.now(clock),
```
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2stubs/controllers/UKTRSubmitController.scala:55]: does this return a date time in the same format as in L71 and L114?

File: app/uk/gov/hmrc/pillar2stubs/controllers/UKTRSubmitController.scala [hunk 2]
```diff
@@ -45,41 +46,51 @@ class UKTRSubmitController @Inject() (
...
+              )
+            } else {
+              submission match {
+                case d @ UKTRSubmissionData(_, _, _, _, _) =>
+                  if (d.liabilities.liableEntities.isEmpty) {
+                    logger.error("Liable entities array is empty in liability submission")
+                    Future.successful(
+                      UnprocessableEntity(
+                        Json.obj(
+                          "errors" -> Json.obj( //2024-12-20T12:54:43.78Z
```
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2stubs/controllers/UKTRSubmitController.scala:70]: do we need this comment?

File: app/uk/gov/hmrc/pillar2stubs/models/UKTRSubmission.scala [hunk 3]
```diff
@@ -26,6 +26,9 @@ sealed trait UKTRSubmission {
   val obligationMTT:        Boolean
   val electionUKGAAP:       Boolean
   val liabilities:          Liability
+
+  def accountingPeriodValid: Boolean =
+    accountingPeriodFrom.isBefore(accountingPeriodTo)
```
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2stubs/models/UKTRSubmission.scala:31]: shouldn't this be implemented in ET stub too?

Comment (ridow-hmrc) [reply app/uk/gov/hmrc/pillar2stubs/models/UKTRSubmission.scala:31]: this should be a validation rule anyway, so should be implemented eventually

**Review by MoDweik — APPROVED** [review]

**Pull request merged**